csi/util.go

```diff
@@ -152,3 +164,24 @@ func detectFileSystem(devicePath string) (string, error) {
 	}
 	return strings.Trim(strings.TrimPrefix(strings.TrimSpace(items[2]), "TYPE="), "\""), nil
 }
+
+func getStatistics(volumePath string) (*volumeStatistics, error) {
+	var statfs *unix.Statfs_t
+	// See http://man7.org/linux/man-pages/man2/statfs.2.html for details.
+	err := unix.Statfs(volumePath, statfs)
```

Comment:
I made a fundamental mistake here which costs me a few hours of debugging. I should have allocated a struct `var statfs unix.Statfs_t` instead of a pointer. Because I only allocated a pointer and the initial value of the pointer is `nil`,  the `unix.Statfs` failed to dereference the `nil` pointer.

Reply:
Your assessment is correct, the unix.Statfs function expects a pointer to a previously allocated struct, which it will end up filling.